components/form/GenerateFormItem.razor.cs

```diff
@@ -0,0 +1,215 @@
+
+    [Parameter] public Func<PropertyInfo, TModel, bool> NotGenerate { get; set; }
+
+    protected override void OnInitialized()
+    {
+        _propertyInfos = typeof(TModel).GetProperties();
+
+        base.OnInitialized();
+    }
+
+    protected override void BuildRenderTree(RenderTreeBuilder builder)
+    {
+        foreach (var property in _propertyInfos)
+        {
+            // Not Generate
+            if (NotGenerate != null && NotGenerate.Invoke(property, (TModel)Form.Model))
+            {
+                continue;
+            }
+
+            var displayName = property.GetCustomAttribute<DisplayNameAttribute>()?.DisplayName ?? property.Name;
+            var isRequired = RulesRequireByType is true && CheckTypeRequire(property);
+            var underlyingType = THelper.GetUnderlyingType(property.PropertyType);
+            RenderFragment? childContent = null;
+            if (Definitions != null)
+            {
+                childContent = Definitions.Invoke(property, (TModel)Form.Model);
+            }
+
+            childContent ??= GenerateByType(underlyingType, property, displayName, isRequired);
+
+            // Make FormItem 
+            if (isRequired)
+            {
+                var formValidation = GetFormValidationRuleByType(underlyingType);
+                builder.OpenComponent(0, typeof(FormItem));
+                builder.AddAttribute(1, "Label", displayName);
+                builder.AddAttribute(2, "Rules", formValidation);
+                builder.AddAttribute(3, "ChildContent", childContent);
+                builder.CloseComponent();
+            }
+            else
+            {
+                builder.OpenComponent(0, typeof(FormItem));
+                builder.AddAttribute(1, "Label", displayName);
+                builder.AddAttribute(2, "ChildContent", childContent);
+                builder.CloseComponent();
+            }
+        }
+    }
+
+    private bool CheckTypeRequire(PropertyInfo property)
+    {
+        if (property.PropertyType.IsValueType)
+        {
+            return !THelper.IsTypeNullable(property.PropertyType);
+        }
+
+        return !(property.GetMethod?.CustomAttributes.Where(x =>
+            x.AttributeType.Name == "NullableContextAttribute") ?? Array.Empty<CustomAttributeData>()).Any();
+    }
+
+    private RenderFragment? GenerateByType(Type underlyingType, PropertyInfo property, string displayName,
+        bool isRequired)
+    {
+        if (underlyingType == typeof(string))
+        {
+            return MakeInputString(property, displayName);
+        }
+
+        if (THelper.IsNumericType(underlyingType) && !underlyingType.IsEnum)
+        {
+            return MakeInputNumeric(property, displayName);
+        }
+
+        if (underlyingType == typeof(DateTime))
+        {
+            return MakeDatePicker(property, isRequired);
+        }
+
+        if (underlyingType.IsEnum)
+        {
+            return MakeEnumSelect(property);
+        }
+
```

Comment:
I think we also need to consider nested objects.

Reply:
I’ll think about how to handle this.